jupyter/d2l-java/chapter_convolutional-modern/densenet.ipynb

```diff
@@ -0,0 +1,632 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "net\n",
+    "    .add(BatchNorm.builder().build())\n",
+    "    .add(Activation::relu)\n",
+    "    .add(Pool.globalAvgPool2dBlock())\n",
+    "    .add(Linear.builder().setUnits(10).build());"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data Acquisition and Training\n",
+    "\n",
+    "Since we are using a deeper network here, in this section, we will reduce the input height and width from 224 to 96 to simplify the computation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "int batchSize = 256;\n",
+    "float lr = 0.1f;\n",
+    "int numEpochs = 10;\n",
+    "\n",
+    "double[] trainLoss;\n",
+    "double[] testAccuracy;\n",
+    "double[] epochCount;\n",
+    "double[] trainAccuracy;\n",
+    "\n",
+    "epochCount = new double[numEpochs];\n",
+    "\n",
+    "for (int i = 0; i < epochCount.length; i++) {\n",
+    "    epochCount[i] = (i + 1);\n",
+    "}\n",
+    "\n",
+    "FashionMnist trainIter =\n",
+    "                FashionMnist.builder()\n",
+    "                        .optPipeline(new Pipeline().add(new Resize(96)).add(new ToTensor()))\n",
+    "                        .optUsage(Dataset.Usage.TRAIN)\n",
+    "                        .setSampling(batchSize, true)\n",
+    "                        .build();\n",
+    "\n",
+    "FashionMnist testIter =\n",
+    "                FashionMnist.builder()\n",
+    "                        .optPipeline(new Pipeline().add(new Resize(96)).add(new ToTensor()))\n",
+    "                        .optUsage(Dataset.Usage.TEST)\n",
+    "                        .setSampling(batchSize, true)\n",
+    "                        .build();\n",
+    "\n",
+    "trainIter.prepare();\n",
+    "testIter.prepare();\n",
+    "\n",
+    "Model model = Model.newInstance(\"cnn\");\n",
+    "model.setBlock(net);\n",
+    "\n",
+    "Loss loss = Loss.softmaxCrossEntropyLoss();\n",
+    "\n",
+    "LearningRateTracker lrt = LearningRateTracker.fixedLearningRate(lr);\n",
+    "Optimizer sgd = Optimizer.sgd().setLearningRateTracker(lrt).build();\n",
+    "\n",
+    "DefaultTrainingConfig config = new DefaultTrainingConfig(loss).optOptimizer(sgd) // Optimizer (loss function)\n",
+    "                .addEvaluator(new Accuracy()) // Model Accuracy\n",
+    "                .addTrainingListeners(TrainingListener.Defaults.logging()); // Logging\n",
+    "\n",
+    "Trainer trainer = model.newTrainer(config);\n",
+    "trainer.initialize(new Shape(1, 1, 96, 96));"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "public void trainingChapter6(ArrayDataset trainIter, ArrayDataset testIter,\n",
```

Comment:
Not really for this PR, but we should consider moving this utility function out of these chapters instead of repeating it (shows up for the first usage in the book, otherwise reuse it from a utility class)